backend/fms_core/resources/project_link_sample.py

```diff
@@ -0,0 +1,104 @@
+import reversion
+from import_export.fields import Field
+from import_export.widgets import ForeignKeyWidget
+from django.core.exceptions import ValidationError, ObjectDoesNotExist
+from ._generic import GenericResource
+from ._utils import skip_rows, add_columns_to_preview
+from ..models import Project, Sample, Container, SampleByProject
+from ..utils import get_normalized_str
+
+
+__all__ = ["ProjectLinkSampleResource"]
+
+ADD_ACTION = "ADD"
+REMOVE_ACTION = "REMOVE"
+
+class ProjectLinkSampleResource(GenericResource):
+    action = Field(column_name='Action')
+    project_name = Field(column_name='Project Name') 
+    sample_name = Field(column_name='Sample Name')
+    barcode = Field(column_name='Sample Container Barcode',
+                     widget=ForeignKeyWidget(Container, 'barcode'))
+    coordinates = Field(column_name='Sample Container Coord')
+
+    class Meta:
+        model = SampleByProject
+        export_order = ('action', 'project_name', 'sample_name', 'barcode', 'coordinates',)
+
+    def before_import(self, dataset, using_transactions, dry_run, **kwargs):
+        skip_rows(dataset, 7)
+
+    def import_obj(self, obj, data, dry_run, **kwargs):
+        errors = {}
+        # Get field info
+        try:
+            super().import_obj(obj, data, dry_run, **kwargs)
+        except ValidationError as e:
+            errors.update(e.update_error_dict(errors).copy())
+        # Get sample object
+        try:
+            obj.sample = Sample.objects.get(name=data["Sample Name"],
+                                            container__barcode=data["Sample Container Barcode"],
+                                            coordinates=data["Sample Container Coord"])
+        except Sample.DoesNotExist:
+            obj.sample = None
+            errors["sample"] = ValidationError([f"No sample with name [{data['Sample Name']}] at those container coordinates."], code="invalid")
+        # Get project object
+        try:
+            obj.project = Project.objects.get(name=data["Project Name"])
+        except Project.DoesNotExist:
+            obj.project = None
+            errors["project"] = ValidationError([f"No project with name [{data['Project Name']}]"], code="invalid")
+
+        # Set object for deletion if REMOVE_ACTION
+        if SampleByProject.objects.filter(sample=obj.sample, project=obj.project).exists():
+            # Perform the desired action
+            if data["Action"] == REMOVE_ACTION:
+                    obj.deleted = True
```

Comment:
too much indentation

Reply:
From my understanding here, you want to set the deleted attribute of that ProjectLinkSample to True. I wonder if there exists any sorts of filters which would prevent attributes with 'deleted == True' to show up in search and in lists?

Reply:
EDIT: While reading the code down there, now I understand that it is a marker that you put on your resource object to know if it is to be deleted or not later in the code. For it to be more explicit, I suggest you rename 'deleted' by 'to_be_deleted' to indicate it's a marker.

Reply:
I would rename the obj."deleted" to something that indicates that it is only a marker for future handling (down in the code). Maybe something like "to_be_deleted" would emphasis that it is not the resource own deletion status, but a marker?